### PR TITLE
Closes #527: Move repeatable code from FittingOrgWordsMatchVoter and FittingAffOrgWordsMatchVoter to a separate class

### LINKED ISSUE #527: Move repeatable code from FittingOrgWordsMatchVoter and FittingAffOrgWordsMatchVoter to a separate class
Remember to change the tests accordingly.

### PR DESCRIPTION
**Pull request by lukdumi**

### CODE REVIEW

File: iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/match/voter/CommonWordsVoter.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    public enum RatioRelation {/** the ratio of common words will be calculated with regard to the number of words in {@link AffMatchAffiliation#getOrganizationName()}*/ 
+                               WITH_REGARD_TO_AFF_WORDS,
+                               /** the ratio of common words will be calculated with regard to the number of words in organization name in an organization object */
+                               WITH_REGARD_TO_ORG_WORDS}
+    
+    private StringFilter stringFilter = new StringFilter();
+    
+    private CommonSimilarWordCalculator commonSimilarWordCalculator; 
+    
+    private RatioRelation ratioRelation = null;//RatioRelation.WITH_REGARD_TO_ORG_WORDS;
```
Comment (madryk) [inline iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/match/voter/CommonWordsVoter.java]: Irrelevant comment?

File: iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/match/voter/CommonAffSectionWordsVoterTest.java [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        
+        // execute
+        
+        new CommonAffSectionWordsVoter(of(), 2, 1.1);
+        
+    }
+    
+    
+    @Test(expected = IllegalArgumentException.class)
+    public void constructor_TOO_LOW_MIN_COMMON_WORDS_PERCENTAGE() {
```
Comment (madryk) [inline iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/match/voter/CommonAffSectionWordsVoterTest.java]: `TOO_LOW_MIN_COMMON_WORDS_PERCENTAGE` should actually be a `TOO_LOW_MIN_COMMON_WORDS_RATIO` to be consistent with naming convention

File: iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/match/voter/CommonSimilarWordCalculatorTest.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+/**
+* @author Łukasz Dumiszewski
+*/
+@RunWith(MockitoJUnitRunner.class)
+public class CommonSimilarWordCalculatorTest {
+
+    private double MIN_WORD_SIMILARITY = 0.83;
+    
+    @InjectMocks
+    private CommonSimilarWordCalculator calculator = new CommonSimilarWordCalculator(0.83);
```
Comment (madryk) [inline iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/match/voter/CommonSimilarWordCalculatorTest.java]: Use in constructor the constant defined above?

File: iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/match/voter/CommonWordsVoterTest.java [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+        
+        // execute
+        
+        new CommonWordsVoter(of(), 2, 1.1, WITH_REGARD_TO_AFF_WORDS);
+        
+    }
+    
+    
+    @Test(expected = IllegalArgumentException.class)
+    public void constructor_TOO_LOW_MIN_COMMON_WORDS_PERCENTAGE() {
```
Comment (madryk) [inline iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/match/voter/CommonWordsVoterTest.java]: `TOO_LOW_MIN_COMMON_WORDS_RATIO`?

File: iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/match/voter/JaroDistanceTest.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        
+        CommonSimilarWordCalculator calculator = new CommonSimilarWordCalculator(0.9);
+        
+        List<String> affWords = Lists.newArrayList("universite", "paris", "curie", "marie", "laboratoire", "oceanographie", "villefranche", "universites", "sorbonne", "pierre");
+        List<String> orgWords = Lists.newArrayList("universitetet", "bergen");
+        
+        System.out.println(calculator.calcSimilarWordNumber(affWords, orgWords));
+        System.out.println(calculator.calcSimilarWordRatio(affWords, orgWords) + "\n");
+        System.out.println(calculator.calcSimilarWordNumber(orgWords, affWords));
+        System.out.println(calculator.calcSimilarWordRatio(orgWords, affWords));
```
Comment (madryk) [inline iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/match/voter/JaroDistanceTest.java]: Probably we could assert these values

File: iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/match/voter/JaroDistanceTest.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    public void test() {
+        
+        for (int i = 0 ; i < 1000; i++) {
+            String a = RandomStringUtils.random(i % 50);
+            String b = RandomStringUtils.randomAlphanumeric(i % 33);
+            
+            System.out.println("a="+a+", b="+b);
+            assertEquals(StringUtils.getJaroWinklerDistance(a, b), StringUtils.getJaroWinklerDistance(b, a), 0.0000001);
+        }
+    }*/
```
Comment (madryk) [inline iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/match/voter/JaroDistanceTest.java]: Is this commented code will be used in the future?

File: iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/match/voter/CommonWordsVoter.java (hunk 1 above)
Comment (lukdumi) [inline iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/match/voter/CommonWordsVoter.java]: well, yes :-)

File: iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/match/voter/CommonAffSectionWordsVoterTest.java (hunk 2 above)
Comment (lukdumi) [inline iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/match/voter/CommonAffSectionWordsVoterTest.java]: :+1:

File: iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/match/voter/CommonSimilarWordCalculatorTest.java (hunk 3 above)
Comment (lukdumi) [inline iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/match/voter/CommonSimilarWordCalculatorTest.java]: :+1:

File: iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/match/voter/CommonWordsVoterTest.java (hunk 4 above)
Comment (lukdumi) [inline iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/match/voter/CommonWordsVoterTest.java]: Right

**Pull request closed without merging**